continuationsPlugin/src/test/scala/continuations/CompilerFixtures.scala

```diff
@@ -174,25 +194,298 @@ trait CompilerFixtures { self: FunSuite =>
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val intTpe = c.definitions.IntType
+      val ctxFunctionTpe = c
+        .definitions
+        .asContextFunctionType(c.definitions.FunctionOf(List(suspend.termRef), intTpe, true, false))
+      ValDef(
+        Symbols
+          .newSymbol(
+            c.owner,
+            Names.termName("mySuspend"),
+            Flags.EmptyFlags,
+            ctxFunctionTpe
+          )
+          .asTerm,
+        Literal(Constant(1))
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val nonSuspendingContextFunctionValDef = FunFixture[Context ?=> ValDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val intTpe = c.definitions.IntType
+      val ctxFunctionTpe = c
+        .definitions
+        .asContextFunctionType(c.definitions.FunctionOf(List(intTpe), intTpe, true, false))
+      ValDef(
+        Symbols
+          .newSymbol(
+            c.owner,
+            Names.termName("mySuspend"),
+            Flags.EmptyFlags,
+            ctxFunctionTpe
+          )
+          .asTerm,
+        Literal(Constant(1))
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val zeroArityNonSuspendNonSuspendingDefDef = FunFixture[Context ?=> DefDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      DefDef(
+        Symbols
+          .newSymbol(
+            c.owner,
+            Names.termName("mySuspend"),
+            Flags.EmptyFlags,
+            c.definitions.IntType)
+          .asTerm,
+        List(
+          List(),
+          List(Symbols.newSymbol(c.owner, Names.termName("s"), Flags.Param, suspend.info))),
+        c.definitions.IntType,
+        Literal(Constant(1))
+      )
+    },
+    teardown = _ => ()
+  )
+
+  val zeroAritySuspendSuspendingDefDef = FunFixture[Context ?=> DefDef](
+    setup = _ => {
+      val c = summon[Context]
+      val suspend = Symbols.requiredClass("continuations.Suspend")
+      val continuation = Symbols.requiredModule("continuations.Continuation")
+      val right = Symbols.requiredModule("scala.util.Right")
+
+      val intType = c.definitions.IntType
+
+      val usingSuspend =
+        Symbols.newSymbol(
+          c.owner,
+          Names.termName("x$1"),
+          Flags.union(Flags.GivenOrImplicit, Flags.Param),
+          suspend.info)
+
+      val anonFunc =
+        Symbols.newAnonFun(c.owner, continuation.companionClass.info)
+
+      val continuationVal = Symbols.newSymbol(
+        c.owner,
+        Names.termName("continuation"),
+        EmptyFlags,
+        continuation.companionClass.typeRef.appliedTo(intType)
+      )
+
+      val rightOne =
+        ref(right)
+          .select(Names.termName("apply"))
+          .appliedToTypes(List(c.definitions.ThrowableType, intType))
+          .appliedTo(Literal(Constant(1)))
+
+      val rhs =
+        Apply(
```

Comment:
just a minor, need to decide if we want `inlined` or not because the fixture here (needs to be in `Inlined(...)`) doesn't match the definition [here](https://github.com/47deg/TBD/pull/53/files#diff-120e125acd11ba8a87fb56283fc506e3739f2e28a9bac11688fbd0d0fc4226d9R19) so could be deceiving if try to use it

Reply:
It's strange that it works in the compiler plugin suite. I think these are working because I used them in the detection suites, where it may not care if the continuation is inlined or not, since it is using the symbol name.

Reply:
Agree, we need to add a test case here, good catch @gregorath

Reply:
@gregorath Addressed in https://github.com/47deg/TBD/pull/53/commits/a4ccd1d7a19ecfd29c3d2e627183f9a77957d671

Please mark the conversation as resolved if you agree.